src/main/java/org/spongepowered/api/world/storage/WorldProperties.java

```diff
@@ -0,0 +1,397 @@
+     * automatically if an attempt is made to transfer an entity to the world
+     * using
+     * {@link Entity#transferToWorld(String, com.flowpowered.math.vector.Vector3d)}
+     * .</p>
+     * 
+     * @return Is enabled
+     */
+    boolean isEnabled();
+
+    /**
+     * Gets the name of this world.
+     * 
+     * @return The name
+     */
+    String getWorldName();
+
+    /**
+     * Sets the name of this world.
+     * 
+     * @param name The name
+     */
+    void setWorldName(String name);
+    
+    /**
+     * Gets the UUID of the world.
+     * 
+     * @return The UUID
+     */
+    UUID getUUID();
+
+    /**
+     * Sets this world as enabled. If a world is enabled then it will be loaded
+     * automatically when the server starts.
+     * 
+     * <p>Additionally a world which is enabled but unloaded may be loaded
+     * automatically if an attempt is made to transfer an entity to the world
+     * using
+     * {@link Entity#transferToWorld(String, com.flowpowered.math.vector.Vector3d)}
+     * .</p>
+     * 
+     * @param state The new state
+     */
+    void setEnabled(boolean state);
+
+    /**
+     * Gets the default spawn position of this world.
+     * 
+     * @return The spawn position
+     */
+    Vector3i getSpawnPosition();
+
+    /**
+     * Sets the default spawn position of this world.
+     * 
+     * @param position The spawn position
+     */
+    void setSpawnPosition(Vector3i position);
+
+    /**
+     * Gets the type of the generator for this world.
+     * 
+     * @return The type
+     */
+    GeneratorType getType();
+
+    /**
+     * Sets the type of this world.
+     * 
+     * @param type The type
+     */
+    void setType(GeneratorType type);
+
+    /**
+     * Gets the seed of this world.
+     * 
+     * @return The seed
+     */
+    long getSeed();
+
+    /**
+     * Sets the seed of this world.
+     * 
+     * @param seed The seed
+     */
+    void setSeed(long seed);
+
+    /**
+     * Gets the number of ticks which have occurred since the world was created.
+     * 
+     * @return The total time in ticks
+     */
+    long getTotalTime();
+
+    /**
+     * Gets the time of day, in ticks. The total number of ticks in a day is
+     * 24000, however this value does not reset to zero at the start of each day
+     * but rather keeps counting passed 24000.
+     * 
+     * @return The time of day
+     */
+    long getWorldTime();
+
+    /**
+     * Sets the time of day, in ticks. The total number of ticks in a day is
+     * 24000, however this value does not reset to zero at the start of each day
+     * but rather keeps counting passed 24000.
+     * 
+     * @param time The time of day
+     */
+    void setWorldTime(long time);
+
+    /**
+     * Gets the Unix time stamp of when this world was last loaded.
+     * 
+     * @return The time this world was last loaded
+     */
+    long getLastTimePlayed();
```

Comment:
@bloodmc put a client annotation on stuff like this there is really no reason for us to not have limited client support in the api